cluster/ci/origin/logging-infra/finished.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"time"
+
+	"os"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+)
+
+type Finished struct {
+	// Timestamp is the time at which we finish
+	// execution, in seconds past the UNIX epoch
+	Timestamp int64 `json:"timestamp"`
+
+	// Passed is the result of the build
+	Passed bool `json:"passed"`
+}
+
+func UploadFinishedData(processLog, metadataFile, artifactDir string, passed bool, gcsBucket *storage.BucketHandle) error {
+	metadata, err := loadMetadata(metadataFile)
+	if err != nil {
+		return err
+	}
+
+	finishedFile, err := writeFinishedMetadata(passed)
+	if err != nil {
+		return err
+	}
+
+	gcsPath := metadata.GcsPath()
+	uploadTargets := map[string]string{
+		finishedFile: gcsPath + "/finished.json",
+		// TODO(skuznets): we want to stream this log during the run
```

Comment:
Right, the TODO you are commenting on is to support that. IDK, having two very very closely related tasks (viewing a job log while it's running and viewing a job log once it's done) in two separate components (deck.gubernator) feels like it would be annoying to use